communicate/communicate.go

```diff
@@ -0,0 +1,34 @@
+package communicate
+
+import (
+	"net/http"
+
+	"../dataloader"
+	"../defination"
+	"../handleerror"
+)
+
+const (
+	getPostsEndpoint    = "https://my-json-server.typicode.com/typicode/demo/posts"
+	getCommentsEndpoint = "https://my-json-server.typicode.com/typicode/demo/comments"
+)
+
+func HandleFunc(writer http.ResponseWriter, request *http.Request, isJSON bool) {
```

Comment:
this HandleFunc depends tightly with your defined dependency.
To apply DI further, you can create `handerImpl struct` that contain all dependencies that can be injected.

The separate dependency initialization out of logic/handler...